### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by AlyAtalla**
set up React Testing Library.
Write unit tests for the files operate.js and calculate.js using Jest.
Create unit tests for all React components:
Use Jest snapshots to test the components.
Use React Testing Library to simulate user interaction.
Run  full tests suite using Jest. All tests pass.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @AlyAtalla,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Your coding partner must contribute to this project. Kindly see to that.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import calculate from './calculate';
+
+describe('calculate', () => {
+  it('resets the calculator object when AC button is pressed', () => {
+    const obj = { total: '3', next: '2', operation: '+' };
+    const buttonName = 'AC';
+    expect(calculate(obj, buttonName)).toEqual({ total: null, next: null, operation: null });
+  });
+});
```
Comment (FrederickMih) [inline src/logic/calculate.test.js]: - [x] Hi @AlyAtalla, based on the requirements of this project, all the functionalities and scenarios of the calculate.js modules should be covered using Jest. For this specific, kindly make sure other scenarios like `buttonName === '0',  buttonName === '.', buttonName === '=', buttonName === '+/-'` are equally covered.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @AlyAtalla 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- All written tests are running successfully without errors :+1: 


## Required Changes ♻️

- [x] App compiled with some console errors. Kindly make sure app is running without errors.

Screenshot from 2023-08-15 10-54-30

_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @AlyAtalla and @2004-okari,

✔️ Well-written and passing tests

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**